### PR TITLE
Updated the gitignore to match Capital Framework gh-pages

### PR DESCRIPTION
**Pull request by jimmynotjim**
Matching Capital Framework with the current best version of `.gitignore` for Jekyll projects.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,13 +68,9 @@ coverage.xml
...
 src/vendor/
-src/static/fonts/
-
-# Jekyll #
-##########
-_site/
-
-# Ruby #
-########
-Gemfile.lock
```
Comment (marteki) [inline .gitignore]: I've had issues in the past on this repo when I came back to it after a while, but my `Gemfile.lock` file was old and out of date. It was keeping me pegged to several versions behind on gems.

File: .gitignore [hunk 2]
```diff
@@ -37,13 +68,9 @@ coverage.xml
 node_modules/
 bower_components/
 .grunt/
+vendor/
+static/vendor/
 src/vendor/
-src/static/fonts/
```
Comment (marteki) [inline .gitignore]: Do we want to have `src/static/fonts/` tracked in Git?

**Review by marteki — CHANGES_REQUESTED** [review]
I added one disagreement with a line removal, and one question about another line removal. I agree with the general idea of updating to match across Capital Framework and the Design Manual!

File: .gitignore (hunk 2 above)
Comment (jimmynotjim) [reply .gitignore]: This doesn't exist, the fonts are in `static/fonts`, no `src/`

File: .gitignore (hunk 1 above)
Comment (jimmynotjim) [reply .gitignore]: That was a mistake, updated with the correct settings and it's been moved under the `Jekyll` heading.

**Review by Scotchester — APPROVED** [review]
:+1:

**Review by marteki — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (Scotchester) [reply .gitignore]: You were incorrect about that. They start in `src/static/fonts` and get copied to `static/fonts`. That said, they should _really_ be originating from `node_modules/cf-icons/src/fonts`.

Comment (jimmynotjim) [reply .gitignore]: @Scotchester I don't have them in my `/src/static/fonts`, am I missing something?

Comment (Scotchester) [reply .gitignore]: I have them locally there. Not sure why they got removed from that location/ignored. I'll look back at the `.gitignore` blame in a sec. But if you look at `grunt:copy`, that was the original intent.

Comment (jimmynotjim) [reply .gitignore]: Yeah, I saw that too. I think what happened was we stopped installing from bower to `/src`, but never updated the tasks to match. That said, we could really use a clean up of the tasks and should probably port them to Gulp so we're consistent with other projects. What say you?

Comment (Scotchester) [reply .gitignore]: [I did it.](https://github.com/cfpb/design-manual/commit/c303bf93a34f3d45023e8a84a5b514eacf64500c) _( 🎶 Do you think I've gone too far?)_

Not sure why. Anyway, I'll fix it by bringing them in from cf-icons.

Comment (Scotchester) [reply .gitignore]: Ah yeah, it probably was the deprecation of Bower.

Comment (jimmynotjim) [reply .gitignore]: Ok, delete that dir from your local before you check them in accidentally ;)

**Pull request merged**